tools/c7n_gcp/c7n_gcp/resources/sql.py

```diff
@@ -68,3 +68,34 @@ def get_resource_params(self, model, resource):
         return {'project': project,
                 'instance': instance,
                 'body': {'settings': {'activationPolicy': 'NEVER'}}}
+
+
+@resources.register('sql-database')
+class SqlDatabase(ChildResourceManager):
+
+    def _get_parent_resource_info(self, resource_info):
+        return {
+            'name': resource_info['instance'],
+            'project': resource_info['project']
+        }
+
+    class resource_type(TypeInfo):
+        service = 'sqladmin'
+        version = 'v1beta4'
+        component = 'databases'
+        enum_spec = ('list', "items[]", None)
+        id = 'name'
+        parent = ('sql-instance', 'name', 'instance')
+
+        @staticmethod
+        def get(client, resource_info):
+            parent_annotation_key = SqlDatabase.get_parent_annotation_key()
+            instance = resource_info[parent_annotation_key]
```

Comment:
this whole method impl looks odd, resource_info isn't a database instance, its a query dictionary typically extracted from an event, it wouldn't have a parent annotation key.

Reply:
Agree. Thanks